src/main/scala/monitors/TLMonitorInjToDcache.scala

```diff
@@ -0,0 +1,89 @@
+// See LICENSE.SiFive for license details.
+
+package freechips.rocketchip.monitors
+
+import chisel3.aop.injecting._
+
+import chisel3.experimental.{BaseModule, FixedPoint}
+import chisel3.{RawModule, VecInit}
+
+import chisel3.aop.injecting.InjectingAspect
+import chisel3.aop._
+import chisel3._
+import firrtl.options.TargetDirAnnotation
+import freechips.rocketchip.config.Parameters
+import freechips.rocketchip.config._
+import freechips.rocketchip.rocket._
+import freechips.rocketchip.stage.{ConfigsAnnotation, TopModuleAnnotation}
+//import freechips.rocketchip.system.{RocketChipStage, TestHarness}
+import freechips.rocketchip.system._
+import chisel3.internal.firrtl._
+//import freechips.rocketchip.util.HasRocketChipStageUtils
+import freechips.rocketchip.util.SelectDiplomacy
+import freechips.rocketchip.tilelink._
+import freechips.rocketchip.formal._
+import freechips.rocketchip.diplomacy._
+import scala.reflect.ClassTag
+
+/** API to access Diplomacy Node data:
+  *  {{{
+  *     val baseNodes = SelectDiplomacy.collectBaseNodes()
+  *     val impMod = SelectDiplomacy.collectImpModules()
+  *     val lazyMod = SelectDiplomacy.collectLazyModules()
+  * 
+  *     val mixedNodes = SelectDiplomacy.mixedNodes()
+  *     val mixedCustomNodes = SelectDiplomacy.mixedCustomNodes()
+  *     val customNodes = SelectDiplomacy.customNodes()
+  *     val junctionNodes = SelectDiplomacy.junctionNodes()
+  *     val mixedAdapterNodes = SelectDiplomacy.mixedAdapterNodes()
+  *     val adapterNodes = SelectDiplomacy.adapterNodes()
+  *     val identityNodes = SelectDiplomacy.identityNodes()
+  *     val ephemeralNodes = SelectDiplomacy.ephemeralNodes()
+  *     val mixedNexusNodes = SelectDiplomacy.mixedNexusNodes()
+  *     val nexusNodes = SelectDiplomacy.nexusNodes()
+  *     val sourceNodes = SelectDiplomacy.sourceNodes()
+  *     val sinkNodes = SelectDiplomacy.sinkNodes()
+  *     val mixedTestNodes = SelectDiplomacy.mixedTestNodes()
+  * 
+  *     val csNode = getSrcNodes[DCache]()
+  *     val cmNode = getManagementNodes[DCache()
```

Comment:
These access functions are fine, but I'm not certain they are powerful enough. We want the user to be able to express queries like:

1. Select the link between CPU and L1, and inject a monitor into the CPU
1. Select the link between CPU and L1, and inject a monitor into the L1
1. Select all sink nodes, and instantiate a TLMonitor on each (don't care which module it is injected to)

Can you show concrete examples demonstrating each of the above queries?